internal/cmd/agent/register/register.go

```diff
@@ -105,6 +105,13 @@ func tryRegister(ctx context.Context, namespace, clusterID string, cfg *rest.Con
 	}, nil
 }
 
+// coreInterface is a subset of corecontrollers.Interface
+type coreInterface interface {
```

Comment:
I changed this to avoid using a full `Interface`, only specifying the APIs that are used. This makes testing easier, as seen below (avoiding to generate the `Interface` mock)